core/state_transition.go

```diff
@@ -386,6 +386,11 @@ func (st *StateTransition) TransitionDb() (*ExecutionResult, error) {
 	if vmerr == nil && msg.To() != nil && *msg.To() == common.HexToAddress(GetPrioritisedFTSOContract(timestamp)) {
 		nominalGasUsed := uint64(21000)
 		nominalGasPrice := uint64(225_000_000_000)
+
+		if st.evm.ChainConfig().IsFlareHardFork1(st.evm.Context.Time) {
```

Comment:
Did you check that this corresponds to the timestamp in the block that is being processed?

Reply:
Yes, data from the block is set into evm context.
```
blockContext := NewEVMBlockContext(header, p.bc, nil)
vmenv := vm.NewEVM(blockContext, vm.TxContext{}, statedb, p.config, cfg)
```